src/typ.rs

```diff
@@ -3901,3 +4228,39 @@ pub fn tuple(elems: Vec<Arc<Type>>) -> Arc<Type> {
 pub fn fn_(args: Vec<Arc<Type>>, retrn: Arc<Type>) -> Arc<Type> {
     Arc::new(Type::Fn { retrn, args })
 }
+
+pub fn binary() -> Arc<Type> {
+    Arc::new(Type::App {
+        args: vec![],
+        public: true,
+        name: "Binary".to_string(),
+        module: vec![],
+    })
+}
+
+pub fn bitstring() -> Arc<Type> {
```

Comment:
What are these other types for? Would they not all be `Binary`?

Reply:
They could be, though I think I would probably make them all a bitstring rather than a binary if we went that way. I'm not sure if they have much use as types outside the bitstring itself. It would certainly be simpler to just have the one.

Reply:
After your explanation of the difference I think bitstring makes more sense than binary as the general type. We could add more specialised types later if we think it'll be useful, for now having a smaller API it's my preference.

Reply:
Having typed my explanations of the difference I came to the same conclusion!